### PR TITLE
Update: show hyperdrive version

### LINKED ISSUE #53: Dashboard: show hyperdrive version
This can be added as a small label somewhere that indicates the hyperdrive version. The version can change if the original hyperdrive is updated, so we need to update this value on every `update` event.

### PR DESCRIPTION
**Pull request by dpaez**
This PR adds a new component: DriveInfo which is a dialog that shows the drive version and the `index.json` (if present). 
There is also a new API endpoint: `api/drives/key/info` used to extract this info. 
Finally, there is a new low level method `info`, added to our `drive` abstraction that collects this information.

Closes #53

### CODE REVIEW

File: packages/dashboard/src/components/DriveInfo.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import CloseIcon from '@material-ui/icons/Close'
+import Typography from '@material-ui/core/Typography'
+
+import ReactJson from 'react-json-view'
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    minWidth: '500px'
+  },
+  indexJSON: {
```
Comment (estebanprimost) [inline packages/dashboard/src/components/DriveInfo.js]: Unused entry

File: packages/dashboard/src/components/DriveInfo.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  },
+  closeButton: {
+    position: 'absolute',
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500]
+  }
+})
+
+const DialogTitle = withStyles(styles)((props) => {
```
Comment (estebanprimost) [inline packages/dashboard/src/components/DriveInfo.js]: Please avoid `withStyles` HOC here. We can use `makeStyles` like before, to be consistent.

File: packages/dashboard/src/components/DriveInfo.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      {onClose ? (
+        <IconButton aria-label='close' className={classes.closeButton} onClick={onClose}>
+          <CloseIcon />
+        </IconButton>
+      ) : null}
+    </MuiDialogTitle>
+  )
+})
+
+const DialogContent = withStyles((theme) => ({
```
Comment (estebanprimost) [inline packages/dashboard/src/components/DriveInfo.js]: Same as before, use `makeStyles`

File: packages/dashboard/src/components/DriveItem.js [hunk 4]
```diff
@@ -49,6 +53,15 @@ const useStyles = makeStyles(theme => ({
   }
 }))
 
+const Fold = (props) =>
```
Comment (estebanprimost) [inline packages/dashboard/src/components/DriveItem.js]: Unnecessary component.

File: packages/dashboard/src/components/DriveItem.js [hunk 5]
```diff
@@ -49,6 +53,15 @@ const useStyles = makeStyles(theme => ({
...
+  (
+    <FoldIcon />
+  )
+
+const Unfold = (props) => (
```
Comment (estebanprimost) [inline packages/dashboard/src/components/DriveItem.js]: Same as before.

File: packages/dashboard/src/components/DriveItem.js [hunk 6]
```diff
@@ -58,7 +71,9 @@ function DriveItem ({ driveKey }) {
   const [downloadedBlocks, setDownloadedBlocks] = useState(0)
   const [files, setFiles] = useState({})
   const [peers, setPeers] = useState([])
+  const [driveInfo, setDriveInfo] = useState({})
   const [showInfo, setShowInfo] = useState(false)
+  const [showDetails, setShowDetails] = React.useState(false)
```
Comment (estebanprimost) [inline packages/dashboard/src/components/DriveItem.js]:
```suggestion
  const [showDetails, setShowDetails] = useState(false)
```

File: packages/dashboard/src/components/DriveItem.js [hunk 7]
```diff
@@ -99,12 +114,18 @@ function DriveItem ({ driveKey }) {
...
+        console.warn(error)
+        return
+      }
       setTitle(drive.key.title)
       setSizeBlocks(drive.size.blocks)
       setSizeBytes(drive.size.bytes)
       setDownloadedBlocks(drive.size.downloadedBlocks)
       setFiles(drive.stats)
       setPeers(drive.peers)
+      setDriveInfo(driveInfo)
```
Comment (estebanprimost) [inline packages/dashboard/src/components/DriveItem.js]:
```suggestion
      setTitle(drive.key.title)
      setSizeBlocks(drive.size.blocks)
      setSizeBytes(drive.size.bytes)
      setDownloadedBlocks(drive.size.downloadedBlocks)
      setFiles(drive.stats)
      setPeers(drive.peers)
      
      const driveInfo = await get(`/drives/${driveKey}/info`)
      if (!response.ok) {
        console.warn(error)
        return
      }
     
      setDriveInfo(driveInfo)
```

Comment (estebanprimost) [reply packages/dashboard/src/components/DriveItem.js]: I just suggest to move `drive` OK data before getting drive info. So we can show at least basic data of drive if first GET is OK.

File: packages/seeder/src/drive.js [hunk 8]
```diff
@@ -33,8 +33,9 @@ class Drive extends EventEmitter {
     this._onUpload = this._onUpload.bind(this)
     this._onUpdate = this._onUpdate.bind(this)
     this._onReady = this._onReady.bind(this)
-
     this._hyperdrive.on('update', this._onUpdate)
+
+    this.getContentAsync = promisify(this._hyperdrive.getContent)
```
Comment (estebanprimost) [inline packages/seeder/src/drive.js]: Unless we want to make `getContentAsync` "visible" from outside (public member), please prefix it with `_`.
```suggestion
    this._getContentAsync = promisify(this._hyperdrive.getContent)
```

File: packages/seeder/src/drive.js [hunk 9]
```diff
@@ -124,6 +125,22 @@ class Drive extends EventEmitter {
...
+    }
+    let indexJSON = {}
+    try {
+      indexJSON = JSON.parse(await this._hyperdrive.readFile('index.json', 'utf-8'))
+    } catch (_) {}
+    const version = this._hyperdrive.version
+    return {
+      version,
+      indexJSON
+    }
```
Comment (estebanprimost) [inline packages/seeder/src/drive.js]: Just a little of space :smile: 
```suggestion
    if (!this._ready) {
      return {}
    }
    
    let indexJSON = {}
    
    try {
      indexJSON = JSON.parse(await this._hyperdrive.readFile('index.json', 'utf-8'))
    } catch (_) {}
    
    const version = this._hyperdrive.version
    
    return {
      version,
      indexJSON
    }
```

**Review by estebanprimost — APPROVED** [review]
Awesome @dpaez. Nice work here :rocket:. Just a few comments about minor things.

**Comment by dpaez** [comment]
PR updated

**Review by estebanprimost — APPROVED** [review]

**Pull request merged**